roachprod: Azure FindActiveAccount fails on non-email username response

Roachprod crashes when it encounters an Azure username that is not in email form. Or that is the suspected cause of the panic that happens in the Azure Provider.

```bash
panic: runtime error: slice bounds out of range [:-1]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/roachprod/vm/azure.(*Provider).FindActiveAccount(0x101b27100?, 0xc000957c20?)
	github.com/cockroachdb/cockroach/pkg/roachprod/vm/azure/azure.go:436 +0x2d3
github.com/cockroachdb/cockroach/pkg/roachprod/vm.FindActiveAccounts.func1({0x1023e9220, 0xc0004c2000})
	github.com/cockroachdb/cockroach/pkg/roachprod/vm/vm.go:520 +0x3b
github.com/cockroachdb/cockroach/pkg/roachprod/vm.ForProvider({0x101e5cffe, 0x5}, 0xc0005354b0)
	github.com/cockroachdb/cockroach/pkg/roachprod/vm/vm.go:551 +0x74
github.com/cockroachdb/cockroach/pkg/roachprod/vm.ProvidersSequential(...)
	github.com/cockroachdb/cockroach/pkg/roachprod/vm/vm.go:573
github.com/cockroachdb/cockroach/pkg/roachprod/vm.FindActiveAccounts(0xc0001ab080?)
	github.com/cockroachdb/cockroach/pkg/roachprod/vm/vm.go:519 +0x30b
github.com/cockroachdb/cockroach/pkg/roachprod.verifyClusterName(0x640001d173?, {0x7ff7bfeff9a9, 0xc}, {0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/roachprod/roachprod.go:83 +0x19c
github.com/cockroachdb/cockroach/pkg/roachprod.Create({0x1023cfaf0, 0xc0001a2018}, 0xc0001ab080, {0x0?, 0xc000535b18?}, 0x3, {{0x7ff7bfeff9a9, 0xc}, 0xebbdb3ed0000, 0xc00088a8d0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/roachprod/roachprod.go:1307 +0xae
main.glob..func1(0x37?, {0xc000672120?, 0xc000535b30?, 0x1001c16a5?})
	main/pkg/cmd/roachprod/main.go:137 +0xc5
main.wrap.func1(0x1036b1b60?, {0xc000672120?, 0x9?, 0x9?})
	main/pkg/cmd/roachprod/main.go:76 +0x3c
github.com/spf13/cobra.(*Command).execute(0x1036b1b60, {0xc000672090, 0x9, 0x9})
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x1036b18e0)
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
main.main()
	main/pkg/cmd/roachprod/main.go:1431 +0x725
```

Jira issue: CRDB-30101

--- a/pkg/roachprod/vm/azure/azure.go
+++ b/pkg/roachprod/vm/azure/azure.go
@@ -432,8 +432,9 @@ func (p *Provider) FindActiveAccount(l *logger.Logger) (string, error) {
 		return "", errors.Wrapf(err, "could not decode JWT claims segment")
 	}
 
-	// This is in an email address format, we just want the username.
-	return data.Username[:strings.Index(data.Username, "@")], nil
+	// If this is in an email address format, we just want the username.
+	username, _, _ := strings.Cut(data.Username, "@")
+	return username, nil
 }
 
 // List implements the vm.Provider interface. This will query all

